packages/sku/src/services/packageManager/getPnpmConfigValue.ts

```diff
@@ -0,0 +1,20 @@
+import { execAsync } from '@/utils/execAsync.js';
+
+type PnpmConfigKey = 'public-hoist-pattern' | 'only-built-dependencies';
+
+export const getPnpmConfigValue = async (
+  configKey: PnpmConfigKey,
+): Promise<string[] | null> => {
+  const rawResult = await execAsync(`pnpm config get ${configKey}`);
```

Comment:
There's a little bit of overhead to running this twice during every `configure` invocation, but in my experience it's relatively minimal.